addOns/replacer/src/test/java/org/zaproxy/zap/extension/replacer/ReplacerParamUnitTest.java

```diff
@@ -0,0 +1,182 @@
+                    MatchType.RESP_HEADER_STR,
+                    ReplacerParam.REPORT_TO_REGEX,
+                    true,
+                    ReplacerParam.REPORT_TO_REPLACEMENT,
+                    null,
+                    false);
+    private ReplacerParam param;
+    private ZapXmlConfiguration configuration;
+
+    @BeforeEach
+    void setUp() {
+        param = new ReplacerParam();
+        configuration = new ZapXmlConfiguration();
+    }
+
+    @Test
+    void shouldHaveConfigVersionKey() {
+        // Given / When
+        param.load(configuration);
+        // Then
+        assertThat(param.getConfigVersionKey(), is(equalTo("replacer[@version]")));
+    }
+
+    @ParameterizedTest
+    @ValueSource(strings = {"true", "false"})
+    void shouldLoadConfirmRemoveFromConfig(boolean state) {
+        // Given
+        configuration.setProperty("replacer.confirmRemoveToken", state);
+        // When
+        param.load(configuration);
+        // Then
+        assertThat(param.isConfirmRemoveToken(), is(equalTo(state)));
+    }
+
+    @Test
+    void shouldDefaultConfirmRemoveTrue() {
+        // Given / When
+        param.load(configuration);
+        // Then
+        assertThat(param.isConfirmRemoveToken(), is(equalTo(true)));
+    }
+
+    @Test
+    void shouldHaveExpectedVersionInDefaultConfig() {
+        // Given / When
+        param.load(configuration);
+        // Then
+        assertThat(param.getCurrentVersion(), is(equalTo(1)));
+    }
+
+    @Test
+    void shouldHaveExpectedNumberOfRulesInDefaultConfig() {
+        // Given / When
+        param.load(configuration);
+        // Then
+        assertThat(param.getRules().size(), is(equalTo(4)));
+        ReplacerParamRule newRule = param.getRule(ReplacerParam.REPORT_TO_DESC);
+        assertNotNull(newRule);
+        assertThat(newRule.getDescription(), is(equalTo(ReplacerParam.REPORT_TO_DESC)));
+    }
+
+    @Test
+    void existingRuleShouldNotBeOverwritten() {
+        // Given / When
+        param.load(configuration);
+        configuration = new ZapXmlConfiguration();
+        createConfigWithReportToRule(configuration);
+        param.load(configuration);
+        ReplacerParamRule enabledReportTo = REPORT_TO_RULE;
+        enabledReportTo.setEnabled(true);
+        param.addRule(enabledReportTo);
+        // Then
+        assertThat(param.getRules().size(), is(equalTo(2)));
+        // getRule returns first match (0th)
+        ReplacerParamRule originalRule = param.getRule(ReplacerParam.REPORT_TO_DESC);
+        assertNotNull(originalRule);
+        assertThat(originalRule.getDescription(), is(equalTo(ReplacerParam.REPORT_TO_DESC)));
+        assertThat(originalRule.isEnabled(), is(equalTo(false)));
+        // Added rule (1st) should be enabled
+        assertThat(param.getRules().get(1).isEnabled(), is(equalTo(true)));
+    }
+
+    @Test
+    void shouldHaveExpectedNumberOfRulesAndStillBeVersionedAfterAddingARule() {
```

Comment:
What's this testing exactly? Why wouldn't it be versioned after adding a rule?

Reply:
🤷‍♂️ that was something you mentioned in IRC, I likely misunderstood. I'll drop it

Reply:
The adding to config was preparation for the test (not a test itself).